File: app/src/main/java/com/mml/movies/viewmodels/MoviesVM.java
package com.mml.movies.viewmodels;


import androidx.lifecycle.ViewModel;
import com.mml.movies.network.models.Movies;
import com.mml.movies.room.MovieDao;
import java.util.List;
import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.schedulers.Schedulers;
import javax.inject.Inject;

@HiltViewModel
public class MoviesVM extends ViewModel{

    @Inject MoviesVM(MovieDao movieDao){
        this.movieDao = movieDao;
        allMovies = movieDao.fetchAllMovies();
        bookmarked = movieDao.getBookmarked();
    }

    MovieDao movieDao;
    public Observable<List<Movies>> bookmarked;
    public Observable<List<Movies>> allMovies;

    public void addBookmark(int id){
        movieDao.addBookMark(id).subscribeOn(Schedulers.io()).subscribe();
    }

    public void removeBookmark(int id){
        movieDao.removeBookMark(id).subscribeOn(Schedulers.io()).subscribe();
    }
}

File: app/src/main/java/com/mml/movies/viewmodels/MoviesVM.kt
package com.mml.movies.viewmodels

import android.util.Log
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.mml.movies.room.MovieDao
import com.mml.movies.room.Stats
import dagger.hilt.android.lifecycle.HiltViewModel
import io.reactivex.rxjava3.schedulers.Schedulers
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import javax.inject.Inject

@HiltViewModel
class MoviesVM @Inject constructor(
    val movieDao: MovieDao,
) : ViewModel() {

    var bookmarked = movieDao.getBookmarked()

    var allMovies = movieDao.fetchAllMovies()

    fun addBookmark(id: Int){
        movieDao.addBookMark(id).subscribeOn(Schedulers.io()).subscribe()
    }

    fun removeBookmark(id: Int){
        movieDao.removeBookMark(id).subscribeOn(Schedulers.io()).subscribe()
    }
}
